File: src/test/java/net/tassia/pancake/util/Base64Test.java
package net.tassia.pancake.util;

import net.tassia.pancake.TestUtils;
import org.junit.Assert;
import org.junit.Test;

public final class Base64Test {

	@Test
	public final void testRandomized() {
		for (int i = 0; i < TestUtils.ENTROPY_ITERATIONS; i++) {
			byte[] data = TestUtils.randomData();
			byte[] data2 = Base64.decode(Base64.encode(data));
			Assert.assertArrayEquals(data, data2);
		}
	}

}

File: src/test/java/net/tassia/pancake/util/Base64Test.kt
package net.tassia.pancake.util

import net.tassia.pancake.ITERATIONS
import kotlin.random.Random
import kotlin.test.Test
import kotlin.test.assertTrue

class Base64Test {

	private fun randomData(): ByteArray {
		return Random.nextBytes(Random.nextInt(1, 8))
	}

	@Test
	fun testBase64() {
		repeat(ITERATIONS) {
			val data = randomData()
			val hex = Base64.encode(data)
			val decoded = Base64.decode(hex)

			if (!data.contentEquals(decoded)) {
				assertTrue(false, "Base64 failed for $data")
			}
		}
	}

}
